src/mobilenet.jl

```diff
@@ -0,0 +1,304 @@
+"""
+  This is a utility function for making sure that all layers have a channel size divisible by 8. 
+"""
+function _make_divisible(v, divisor, min_value = nothing)
+  if isnothing(min_value)
+    min_value = divisor
+  end
+  new_v = max(min_value, floor(Int, v + divisor / 2) ÷ divisor * divisor)
+  # Make sure that round down does not go down by more than 10%
+  (new_v < 0.9 * v) ? new_v + divisor : new_v
+end
+
+# MobileNetv2
+
+"""
+    invertedresidualv2(inplanes, outplanes, stride, expand_ratio)
+
+Create a basic inverted residual block for MobileNetv2
+([reference](https://arxiv.org/abs/1801.04381)).
+
+# Arguments
+- `inplanes`: number of input feature maps
+- `outplanes`: number of output feature maps
+- `stride`: stride of the convolutional layer, has to be either 1 or 2
+- `expand_ratio`: ratio of the input feature maps and the inner bottleneck feature maps
+"""
+function invertedresidualv2(inplanes, outplanes, stride, expand_ratio)
+  @assert stride in [1, 2] "`stride` has to be 1 or 2"
+  hidden_planes = floor(Int, inplanes * expand_ratio)
+
+  if expand_ratio == 1
+    invres = Chain(
+      conv_bn((3, 3), hidden_planes, hidden_planes, relu6; bias = false, stride,
+               pad = 1, groups = hidden_planes)...,
+      conv_bn((1, 1), hidden_planes, outplanes, identity; bias = false)...
+    )
+  else
+    invres = Chain(
+      conv_bn((1, 1), inplanes, hidden_planes, relu6; bias = false)...,
+      conv_bn((3, 3), hidden_planes, hidden_planes, relu6; bias = false, stride,
+               pad = 1, groups = hidden_planes)...,
+      conv_bn((1, 1), hidden_planes, outplanes, identity; bias = false)...
+    )
+  end
+
+  (stride == 1 && inplanes == outplanes) ? SkipConnection(invres, +) : invres
+end
+
+"""
+    mobilenetv2(width_mult; nclasses = 1000)
+
+Create a MobileNetv2 model.
+([reference](https://arxiv.org/abs/1801.04381)).
+
+# Arguments
+- `width_mult`: Controls the number of channels in each layer, with 1.0 being the original
+  model as detailed in the paper.
+- `nclasses`: the number of output classes
+"""
+function mobilenetv2(width_mult; nclasses = 1000)
+  configs = [
+    # t, c, n, s
+    [1, 16, 1, 1],
+    [6, 24, 2, 2],
+    [6, 32, 3, 2],
+    [6, 64, 4, 2],
+    [6, 96, 3, 1],
+    [6, 160, 3, 2],
+    [6, 320, 1, 1],
+  ]
+  
+  # building first layer
+  inplanes = _make_divisible(32 * width_mult, width_mult == 0.1 ? 4 : 8)
+  layers = []
+  append!(layers, conv_bn((3, 3), 3, inplanes, stride = 2))
+
+  # building inverted residual blocks
+  for (t, c, n, s) in configs
+    outplanes = _make_divisible(c * width_mult, width_mult == 0.1 ? 4 : 8)
+    for i in 1:n
+      push!(layers, invertedresidualv2(inplanes, outplanes, i == 1 ? s : 1, t))
+      inplanes = outplanes
+    end
+  end
+
+  # building last several layers
+  outplanes = (width_mult > 1.0) ? _make_divisible(1280 * width_mult, width_mult == 0.1 ? 4 : 8) : 1280
```

Comment:
Similarly, should we let the 1280 be customizable?

Reply:
Fixed this in the latest API